carbon-design-system/carbon
# feat(number-input): add option to hide steppers

Closes #8386 

This PR adds a prop (`hideSteppers`) to the `NumberInput` component which can be used to hide the "-" and "+" buttons.

Clickable steppers are great for certain cases of number inputs like small, discrete changes but can be problematic when dealing with large numbers or a very granular step.
Related: https://www.nngroup.com/articles/input-steppers/

The Carbon guidelines also recognize this and discourage its use when dealing with large numbers:
![image](https://user-images.githubusercontent.com/28265588/117842443-3af9b980-b27e-11eb-8194-e10b48aba47f.png)
> https://www.carbondesignsystem.com/components/number-input/usage/#formatting

Though, as of right now the best way to achieve this is by using the `TextInput` component and passing `type="number"` (plus some css styling to get rid of browser steppers). This seems a bit hacky for a recommended variant of the component. Therefore the new prop `hideSteppers` aims to solve this by giving an explicit option to hide the steppers based on the individual case.

#### Changelog

**New**

- Added prop `hideSteppers` to `NumberInput`

#### Testing / Reviewing

- Make sure the steppers aren't rendered when `props.hideSteppers` is passed
- Make sure the invalid and warning validation icons are placed correctly when steppers are hidden
